jsonschema/cli.py

```diff
@@ -9,39 +13,134 @@
+                error=self.PARSING_ERROR_MSG.format(
+                    file_name=file_name,
+                    exception=exception,
+                ),
+            )
+        elif self.output_format == "plain":
+            msg = self.PARSING_ERROR_MSG.format(
+                file_name=file_name,
+                exception=exception,
+            )
+        else:  # pragma: no cover
+            raise ValueError(
+                "Output mode '{}' is unknown by this function"
+                .format(self.output_format)
+            )
+        self.stderr.write(msg)
 
+    def write_valid_error(self, object_name, error_obj):
+        if self.output_format == "pretty":
+            msg = self.PRETTY_ERROR_MSG.format(
+                object_name=object_name,
+                error=error_obj,
+            )
+        elif self.output_format == "plain":
+            msg = self.oneline_format.format(
+                object_name=object_name,
+                error=error_obj,
+            )
+        else:  # pragma: no cover
+            raise ValueError(
+                "Output mode '{}' is unknown by this function"
+                .format(self.output_format)
+            )
+        self.stderr.write(msg)
+
+    def write_valid_success(self, object_name):
+        if self.output_format == "pretty":
+            msg = self.PRETTY_SUCCESS_MSG.format(
+                object_name=object_name
+            )
+        elif self.output_format == "plain":
+            # Nothing to print in plain mode, only errors are wanted.
+            msg = ""
+        else:  # pragma: no cover
+            raise ValueError(
+                "Output mode '{}' is unknown by this function"
+                .format(self.output_format)
+            )
+        self.stdout.write(msg)
+
+
+# ------------------------------------------------------------------------------
+# FUNCTIONS USED BY ARGPARSE
+# ------------------------------------------------------------------------------
 def _namedAnyWithDefault(name):
     if "." not in name:
         name = "jsonschema." + name
     return namedAny(name)
 
 
-def _json_file(path):
-    with open(path) as file:
-        return json.load(file)
-
-
+# ------------------------------------------------------------------------------
+# ARGUMENT PARSING
+# ------------------------------------------------------------------------------
 parser = argparse.ArgumentParser(
     description="JSON Schema Validation CLI",
 )
 parser.add_argument(
     "-i", "--instance",
     action="append",
     dest="instances",
-    type=_json_file,
+    type=str,
     help=(
-        "a path to a JSON instance (i.e. filename.json) "
-        "to validate (may be specified multiple times)"
+        "A path to a JSON instance (i.e. filename.json) "
+        "to validate (may be specified multiple times). "
+        "If this option is not used, an instance can be given through stdin. "
+        "If none is given, only the schema is checked."
```

Comment:
I don't think I follow the last sentence here -- is `none` different from "not given"? And if it means `None`, how does someone get that through from a command line?

Reply:
What I meant was “ If neither -i nor stdin is given, only the schema is checked."
I am changing that.